pkg/trustedcabundle/trustedcabundle.go

```diff
@@ -168,54 +166,41 @@ func ConfigureTrustedCABundle(ctx context.Context, cli client.Client, log logr.L
 	return nil
 }
 
-// when DSCI TrustedCABundle.ManagementState is set to `Managed`.
-func AddCABundleConfigMapInAllNamespaces(ctx context.Context, cli client.Client, dscInit *dsciv1.DSCInitialization) error {
-	namespaceList := &corev1.NamespaceList{}
-	if err := cli.List(ctx, namespaceList); err != nil {
-		return err
-	}
-
-	for i := range namespaceList.Items {
-		ns := &namespaceList.Items[i]
-		// check namespace status if not Active, then skip
-		if ns.Status.Phase != corev1.NamespaceActive {
-			continue
-		}
-
+// when DSCI TrustedCABundle.ManagementState is set to `Managed`, add new configmap into all active namespaces.
+func AddCABundleCMInAllNamespaces(ctx context.Context, cli client.Client, dscInit *dsciv1.DSCInitialization) error {
+	var multiErr *multierror.Error
+	err := cluster.ProcessAllNamespace(ctx, cli, func(ns *corev1.Namespace) error {
 		if ShouldInjectTrustedBundle(ns) {
-			if err := wait.PollUntilContextTimeout(ctx, time.Second*1, time.Second*10, false, func(ctx context.Context) (bool, error) {
+			if pollErr := wait.PollUntilContextTimeout(ctx, time.Second*1, time.Second*10, false, func(ctx context.Context) (bool, error) {
 				if cmErr := CreateOdhTrustedCABundleConfigMap(ctx, cli, ns.Name, dscInit.Spec.TrustedCABundle.CustomCABundle); cmErr != nil {
 					// Logging the error for debugging
 					fmt.Printf("error creating cert configmap in namespace %v: %v", ns.Name, cmErr)
 					return false, nil
 				}
 				return true, nil
-			}); err != nil {
-				return err
+			}); pollErr != nil {
+				multiErr = multierror.Append(multiErr, pollErr)
 			}
 		}
+		return nil
+	})
+	if err != nil {
+		return err
 	}
-	return nil
+	return multiErr.ErrorOrNil()
 }
 
-// when DSCI TrustedCABundle.ManagementState is set to `Removed`.
-func RemoveCABundleConfigMapInAllNamespaces(ctx context.Context, cli client.Client) error {
+// when DSCI TrustedCABundle.ManagementState is set to `Removed`, delete configmap from all active namespaces.
+func RemoveCABundleCMInAllNamespaces(ctx context.Context, cli client.Client) error {
 	var multiErr *multierror.Error
-
-	namespaceList := &corev1.NamespaceList{}
-	if err := cli.List(ctx, namespaceList); err != nil {
-		return multierror.Append(multiErr, err)
-	}
-
-	for i := range namespaceList.Items {
-		ns := &namespaceList.Items[i]
-		// check namespace status if not Active, then skip
-		if ns.Status.Phase != corev1.NamespaceActive {
-			continue
-		}
+	err := cluster.ProcessAllNamespace(ctx, cli, func(ns *corev1.Namespace) error {
 		if err := DeleteOdhTrustedCABundleConfigMap(ctx, cli, ns.Name); err != nil {
 			multiErr = multierror.Append(multiErr, err)
 		}
+		return nil
+	})
+	if err != nil {
+		return err
 	}
 	return multiErr.ErrorOrNil()
```

Comment:
This is a bit odd now, because we might be losing errors that occured in the closure up to this point (in previous pagination chunks). Can be reworked to sth like this and propagate all the errors.

```suggestion
	processErr := cluster.ProcessAllNamespace(ctx, cli, func(ns *corev1.Namespace) error {
		multiErr = multierror.Append(multiErr, DeleteOdhTrustedCABundleConfigMap(ctx, cli, ns.Name))
		
		return nil
	})
	
	return multiErr.Append(multiErr, processErr).ErrorOrNil()
```